### PR TITLE
build: use rules_cc and use buildifier -lint to enforce it

### PR DESCRIPTION
**Pull request by lizan**
Commit Message:
Partially address #9488. Resurrected #10597.

Additional Description:
Risk Level: Low
Testing: CI
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by lizan** [comment]
Note to reviewer: please only review the first commit, the second one is generated by fix_format.

File: tools/code_format/check_format_test_helper.py [hunk 1]
```diff
@@ -61,12 +61,12 @@ def fixFileExpectingSuccess(file, extra_input_files=None):
   command, infile, outfile, status, stdout = fixFileHelper(file,
                                                            extra_input_files=extra_input_files)
   if status != 0:
-    print("FAILED:")
+    print("FAILED:" + infile)
```
Comment (htuch) [inline tools/code_format/check_format_test_helper.py:64]: Nit: add a space after the `:`?

**Comment by lizan** [comment]
@htuch done, I need v2-api LGTM too :)

**Pull request merged**